### PR TITLE
document the static/dynamic behaviour and working with can-import

### LINKED ISSUE #2336: can-import is not working as described in the documentation
simple app:

**main.js**
```
import appTemplate from "src/app.stache!";

$('body').append(appTemplate());
```

**app.stache**
_is not working_
```
<can-import from="testapp/components/welcome/"></can-import>
<welcome></welcome>
``` 

_is not working_
```
<can-import from="testapp/components/welcome/">
    <welcome></welcome>
</can-import>
``` 

_is working_
```
<can-import from="testapp/components/welcome/" />
<welcome></welcome>
```

the documentation shows all 3 typs https://canjs.com/docs/can%7Cview%7Cstache%7Csystem.import.html

you may want to download this testapp repo https://github.com/pYr0x/testapp

### PR DESCRIPTION
**Pull request by pYr0x**
close #2336 and #2320

@matthewp can you have a quick look into? i will merge that if you give the OK

### CODE REVIEW

File: view/stache/doc/can-import.md [hunk 1]
```diff
@@ -48,6 +48,43 @@ Example:
...
+```
+<can-import from="components/foobar">
+  {{#if isResolved}}
+  <foobar/>
+  {{/if}}
+</can-import>
+```
+which is equivalent to a stealJS import like:
+```
+import loader from "@loader";
```
Comment (matthewp) [inline view/stache/doc/can-import.md]: Even though it is more correct to do it this way, I would just use System.import for simplicity.

Comment (pYr0x) [inline view/stache/doc/can-import.md]: ok.. i rewrite to `System.import`. i can leave also a note that `@loader` would be recommended way ;)

**Comment by pYr0x** [comment]
@nlundquist what do you think. all aspects of importing modules are described?

@daffl if this PR is merged can you update the website pls?

File: view/stache/doc/can-import.md [hunk 2]
```diff
@@ -48,6 +48,98 @@ Example:
...
+which is equivalent to a stealJS import like:
+```
+System.import('components/foobar').then(function(foobar) {
+ // access to the module you loaded.
+ // e.g. access to a component's ViewModel 
+ // foobar.ViewModel
+});
+```
+
+Please notice that when dynamically import modules in a stache file, the scope inside `<can-import>` is a Promise, so you have to wait for until it is resolved.
```
Comment (phillipskevin) [inline view/stache/doc/can-import.md]: Maybe this should say 

Please notice that when dynamically import**ing** modules in a stache file, the scope inside `<can-import>` is a Promise, so you have to wait ~~for~~ until it is resolved.

File: view/stache/doc/can-import.md [hunk 3]
```diff
@@ -48,6 +48,98 @@ Example:
...
+Use the `{{#if isResolved}}` helper for that.
+
+The recommended way to progressive loading modules with StealJS is to use the `@loader` module provided by steal.
```
Comment (phillipskevin) [inline view/stache/doc/can-import.md]: The recommended way to progressive**ly** load~~ing~~ modules with StealJS is to use the `@loader` module provided by steal.

File: view/stache/doc/can-import.md [hunk 4]
```diff
@@ -48,6 +48,98 @@ Example:
...
+
+var renderer = canStache('' +
+                '<can-import from="mymodule" {^value}="*module"/>' +
+                'Modules default: {{*module.default}}' +
+                'Modules doMath: {{*module.doMath("2","3")}}');
+                
+$('body').append(renderer());
+```
+
+The other way is to import the module into the templates data.
```
Comment (phillipskevin) [inline view/stache/doc/can-import.md]: Probably should be

template's data

**Comment by phillipskevin** [comment]
👍 I think this really helps clarify the uses.

**Comment by pYr0x** [comment]
thx @phillipskevin for the review... i will merge this.

**Comment by leoj3n** [comment]
Thanks for this; much needed.

**Pull request merged**